### DISCUSSION TITLE
resetPage() not work

### DISCUSSION THREAD

**Discussion by Shadecute** [livewire/livewire#4902] [7. Bugs (archived)] [answered]
i have a query , 
I tried two methods for pagination.
The page reset works in the first method but does not work in the second method.

methods one:
```
    public function render()
    {
        $images = Gallery::query()
            ->when($this->selected, function($query) {
                return $query->where('status', $this->selected);
            })->paginate($this->perPage);
        return view('livewire.dashboard.add-image', compact('images'));
    }
```
I will reset the page using the this function.
```
public function updatedSelected(){
    $this->resetPage();
}
```
in blade:
```
{{ $images->links() }}

```
So far everything is fine and it resets well.

But when I want to use the Infinite Scrolling method The query works but the page does not reset.
 
method 2:
```
class ImageList extends Component
{

    use WithPagination;

    public $perPage = 10;
    public function loadMore()
    {
        $this->perPage += 10;
    }
public function updatedSelected(){
    $this->resetPage();

}
    public function render()
    {
        $images = Gallery::query()
            ->when($this->selected, function($query) {
                return $query->where('status', $this->selected);
            })->paginate($this->perPage);
        return view('livewire.dashboard.image-list', compact('images'));
}
```

in blade:
```
 @if($images->hasMorePages())
        <button wire:click.prevent="loadMore">Load more</button>
    @endif
```
By changing the category of the page can not be reset in the second method.
Why does  in the second  method page reset not work? 
Maybe this is a bug!
please guide me , Thanks.

**Comment by tushargugnani** [comment]
Where are you using reset in the second method?

**Reply by Shadecute** [reply]
Sorry the code was forgotten here. I edited it

**Comment by tushargugnani** [comment, marked as answer]
resetPage resets the page to 1, since in the second approach we are doing everything in the single page, resetPage won't have any effect. So instead of resetPage, you can simply to perPage = 10; to reset the results.

**Reply by Shadecute** [reply]
Thanks for your explanation, I realized that I can not use the reset page in single page mode. But I did not understand how to use your solution.
this: **_you can simply to perPage = 10; to reset the results._**

```
  public $perPage = 10;
    public function loadMore()
    {
        $this->perPage += 10;
    }
```
My code is like this but it does not reset results.

**Reply by tushargugnani** [reply]
```
public function updatedSelected(){
    $this->perPage = 10;
}
```

**Reply by Shadecute** [reply]
Wow thank you, I really could not think of a solution anymore. With your guidance, my problem was solved.
